src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs

```diff
@@ -11,37 +11,46 @@
 namespace Microsoft.CodeAnalysis.Scripting.Hosting
 {
     /// <summary>
-    /// Extends MetadataFileReferenceResolver to enable resolution of assembly
-    /// simple names in the GAC.
+    /// Resolves assembly identities in Global Assembly Cache.
     /// </summary>
     internal sealed class GacFileResolver : IEquatable<GacFileResolver>
     {
         /// <summary>
-        /// Architecture filter used when resolving assembly references.
+        /// Returns true if GAC is available on the platform.
         /// </summary>
-        public ImmutableArray<ProcessorArchitecture> Architectures { get; }
+        public static bool IsAvailable => CoreClrShim.AssemblyLoadContext.Type == null;
```

Comment:
This is a bit indirect.  I'd prefer we have a property somewhere which directly declared: IsRuntimeCoreClr, IsRuntimeDesktop, etc ...